test/common/integration/client_integration_test.cc

```diff
@@ -19,11 +21,17 @@ namespace {
 
 // Based on Http::Utility::toRequestHeaders() but only used for these tests.
 Http::ResponseHeaderMapPtr toResponseHeaders(envoy_headers headers) {
-  Http::ResponseHeaderMapPtr transformed_headers = Http::ResponseHeaderMapImpl::create();
+  std::unique_ptr<Http::ResponseHeaderMapImpl> transformed_headers =
```

Comment:
There's an unfortunate bit of cut/paste from header_utility.cc here. Is there any way to share some code, or is that a non-starter?

Reply:
+1 I am guessing that this direction was not written in prod code because we never go from c->c++ there. But I think we can probably have a shared function that does the transformation in the common map type and wrapper functions for the derived types?

Reply:
yeah, the problem is the base class doesn't have setFormatter, so then we end up having to set in the request/response specific subclasses and assert that work is done in the shared class.  kinda ugly - maybe I'll fix the APIs upstream so I can do a cleanup PR later.